docs/src/mdx-components.tsx

```diff
@@ -165,15 +170,28 @@ export const a = function Anchor({
           line-height: normal;
           /* TextLink's display as inline-flex cause long links to break paragraphs */
           display: inherit;
+          /* Ensure the icon stays inline */
+          span {
+            display: inline;
+            svg {
+              display: inline;
+            }
+          }
         }
       `}
     >
       <TextLink
         href={href}
         external={isExternal}
-        iconRight={isExternal && <NewWindow />}
-        onClick={event => {
+        iconRight={useExternalIcon && <NewWindow ariaLabel="Opens in new window" />}
+        onClick={(event: React.SyntheticEvent<HTMLLinkElement>) => {
           if (isExternal || !href) return;
+
+          // Allows opening in a new tab with ctrl/cmd + click
+          if (isKeyboardEvent(event)) {
+            if (event.metaKey || event.ctrlKey) return;
+          }
+
```

Comment:
There's a way in which can avoid that `onClick` handler. 💡 We can create a component that adjust's Gatsby's Link component to TextLink's `asComponent` prop:

```tsx
function LinkForOrbitTextLink({ href, ...props }: { href: string }) {
  return <Link to={href} {...props} />;
}
```

Then just add a conditional `asComponent` to `TextLink`:

```tsx
<TextLink
  // @ts-expect-error type declaration is not permissive enough
  asComponent={isExternal ? "a" : LinkForOrbitTextLink}
  // ...
```

TypeScript will probably complain because of the dumb type declaration which only allows `className` prop for `asComponent` components, which is why I added the `@ts-expect-error`.

Reply:
Seems interesting. I remembered you saying you were going to use Gatsby's Link and `asComponent`, so I figured there was a reason it wasn't there. But this way seems more elegant.